lib/assessment.rb

```diff
@@ -20,20 +27,36 @@
 # Name and location should be writeable, but age should be read-only.
 
 ##
-# your code here
+class Person
+  def initialize(name, age, location)
+    @name = name
+    @age = age
+    @location = location
+  end
+
+  attr_reader :age
+  attr_writer :name
+  attr_writer :location
```

Comment:
Missed this earlier, but you should be using `attr_accessor` here, not `attr_writer`. The reason is that `attr_writer` only creates a 'setter' method, _not a 'getter' one_. Since wanting to be able to write to a property but not read from it is _extremely_ unusual, `attr_writer` is rarely used.